command.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2016 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package updater
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/keybase/go-logging"
+)
+
+func timerWithTimeout(timeout time.Duration, cmd *exec.Cmd, log logging.Logger) *time.Timer {
+	return time.AfterFunc(timeout, func() {
+		if cmd != nil && cmd.Process != nil {
+			log.Warningf("Command timed out, killing")
+			if err := cmd.Process.Kill(); err != nil {
+				log.Warningf("Error trying to kill process: %s", err)
+			}
+		}
+	})
```

Comment:
If you want it to be async then you can return an error channel instead of an error, and run the entire second half of this function in a goroutine which writes to a buffered error channel.